crates/uv/src/commands/pip_install.rs

```diff
@@ -518,12 +517,12 @@ async fn resolve(
         vec![]
     } else {
         // Combine upgrade and reinstall lists
-        let mut exclusions: HashSet<&PackageName> = if let Reinstall::Packages(packages) = reinstall
-        {
-            HashSet::from_iter(packages)
-        } else {
-            HashSet::default()
-        };
+        let mut exclusions: FxHashSet<&PackageName> =
+            if let Reinstall::Packages(packages) = reinstall {
+                FxHashSet::from_iter(packages)
+            } else {
+                FxHashSet::default()
+            };
```

Comment:
This is going to conflict with #2596 which refactors this already.

Reply:
Reverting since it's not required for the PR.